k-distribution/src/main/scripts/bin/kparse

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env bash
+set -e
+set -o pipefail
+set -u
+
+export PATH="$(cd "$(dirname "$0")"; pwd):$PATH"
+
+# initialize flags
+ARGV=()
+dir=.
+fold_lines="fold -s"
+inputMode=program
+literal=false
+outputFile=-
+result=1
+verbose=false
+
+# setup temp files
+now=`date +"%Y-%m-%d-%H-%M-%S"`
+tempDir="$(mktemp -d .kparse-${now}-XXXXXXXXXX)"
+tempFiles=("$tempDir")
+trap 'rm -rf ${tempFiles[*]}' INT TERM EXIT
+
+execute () {
+  (
+  if $verbose; then
+    set -x
+  fi
+  "$@"
+  )
+}
+
+KPARSE=$(basename "$0")
+KAST_UTIL="$(dirname "$0")/kast"
+
+error () {
+  printf "[Error] Inner Parser: $@\n" | $fold_lines 1>&2
+  exit $result
+}
+
+print_usage () {
+cat <<HERE
+Usage: $KPARSE <INPUT_FILE> [-d DIR] [-m MODULE] [-s SORT]
+Generate an on-the-fly parser from a K definition and use it to parse a term.
+
+Mandatory arguments to long options are mandatory for short options too.
+
+$KPARSE expects exactly 1 positional argument. It must be either a readable
+file, or '-'. By default, this file will be parsed using the main syntax
+module of the K definition and the sort of the \$PGM configuration variable.
+The resulting AST, in KORE format, will be output on stdout. This can be
+customized; see below. If no \$PGM configuration variable exists, the default
+sort will be the KItem sort.
```

Comment:
This is a bit confusing to me. Are you saying just that the output on stdout can be customized? Then I would say "will be output on stdout (or to a file, as specified, see below).". If you mean that the output format AST can be changed (to KAST or something), then I thought that was a non-goal of this tool?